### PR TITLE
CSHARP-4378: add integration tests for Azure KMS

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: evergreen/evergreen.yml [hunk 1]
```diff
@@ -114,7 +114,8 @@ functions:
             # If this was a patch build, doing a fresh clone would not actually test the patch
             cp -R ${PROJECT_DIRECTORY}/ $DRIVERS_TOOLS
           else
-            git clone https://github.com/mongodb-labs/drivers-evergreen-tools.git $DRIVERS_TOOLS
+            #git clone https://github.com/mongodb-labs/drivers-evergreen-tools.git $DRIVERS_TOOLS
```
Comment (DmitryLukyanov) [inline evergreen/evergreen.yml:117]: will be reverted

File: tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs [hunk 2]
```diff
@@ -1970,7 +2023,14 @@ private Exception AssertInnerEncryptionException(Exception ex, Type exType, para
                 }
             }
 
-            e.Should().BeOfType(exType);
+            if (typeof(OperationCanceledException).IsAssignableFrom(exType))
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs:2026]: Maybe it's worth putting a comment with real exception types handled by this case.

File: build.cake [hunk 3]
```diff
@@ -461,6 +461,29 @@ Task("TestCsfleWithMongocryptdNetStandard20").IsDependentOn("TestCsfleWithMongoc
 Task("TestCsfleWithMongocryptdNetStandard21").IsDependentOn("TestCsfleWithMongocryptd");
 Task("TestCsfleWithMongocryptdNet60").IsDependentOn("TestCsfleWithMongocryptd");
 
+Task("TestCsfleWithAzureKms")
+    .IsDependentOn("Build")
+    .DoesForEach(
+        items: GetFiles("./**/*.Tests.csproj"),
+        action: (BuildConfig buildConfig, Path testProject) =>
+    {
+        var settings = new DotNetTestSettings
```
[the remark this answers is not in the document]
Comment (DmitryLukyanov) [reply build.cake:470]: Good point, I also thought about something like this. I will try to do it

[the remark this answers is not in the document]
Comment (DmitryLukyanov) [reply build.cake]: done

**Review by BorisDog — APPROVED** [review]
LGTM (minor comment)

File: build.cake [hunk 4]
```diff
@@ -975,3 +772,25 @@ string[] CreateLoggers()
     var consoleLogger = "console;verbosity=detailed";
     return new []{ junitLogger, consoleLogger };
 }
+
+void RunTests(BuildConfig buildConfig, Path path, string filter = null)
```
[the remark this answers is not in the document]
Comment (DmitryLukyanov) [reply build.cake:776]: discussed offline, we can reconsider it later

**Pull request merged**